lib/ocpp/v201/charge_point.cpp

```diff
@@ -3473,6 +3481,12 @@ DataTransferResponse ChargePoint::data_transfer_req(const DataTransferRequest& r
     ocpp::Call<DataTransferRequest> call(request, this->message_queue->createMessageId());
     auto data_transfer_future = this->send_async<DataTransferRequest>(call);
 
+    if (data_transfer_future.wait_for(DEFAULT_WAIT_FOR_FUTURE_TIMEOUT) == std::future_status::timeout) {
```

Comment:
What happens with the message if we cancel it here. There is no point to send it anymore I would say.

Is it an idea to also provide an interface that just sends the message and the response comes from a callback? Or instead of waiting for the future here, we could return a future that the other side could handle whenever there is a response.

Reply:
There are some reasons why we should still send the message:
* If QueueAllMessages is true
* If the caller doesn't  care about the response from the CSMS

I'm open to discuss other handling for the `send_async` mechansim and I like your suggestion, but I would rather not add this as part of this PR, since this does improve the current implementation. I'd propose to put this on the agenda for the next WG call?